catalog/app/containers/Bucket/PackageDialog/actions.ts

```diff
@@ -0,0 +1,41 @@
+import type * as H from 'history'
+import * as React from 'react'
+import * as RRDom from 'react-router-dom'
+
+import type { ActionPreferences } from 'utils/BucketPreferences'
+
+const Actions: Partial<ActionPreferences> = {
+  revisePackage: true,
+}
+
+type Action = keyof typeof Actions
+
+function clearActions(searchParams: URLSearchParams, history: H.History) {
+  searchParams.delete('action')
+  history.replace({
+    search: searchParams.toString(),
+  })
+}
+
+function isActions(actions: string[]): actions is Action[] {
+  const unsupportedActions = actions.filter((a) => !Actions[a as Action])
+  return !unsupportedActions.length
```

Comment:
probably can be expressed as `actions.every((a) => !!Actions[a as Action])`

Reply:
Noted, will change in next PR